jcl/src/java.base/share/classes/java/lang/invoke/StaticFieldVarHandle.java

```diff
@@ -132,1349 +135,1915 @@
+				return _unsafe.weakCompareAndSwapObjectAcquire(definingClass, varHandle.vmslot, testValue, newValue);
 /*[ENDIF]*/				
 			}
 			
 			private static final boolean weakCompareAndSetRelease(Object testValue, Object newValue, StaticFieldVarHandle varHandle) {
+				Class<?> definingClass = varHandle.definingClass;
+				ensureClassIsInitialized(definingClass);
 /*[IF Sidecar19-SE-B174]*/				
-				return _unsafe.weakCompareAndSetObjectRelease(varHandle.definingClass, varHandle.vmslot, testValue, newValue);
+				return _unsafe.weakCompareAndSetObjectRelease(definingClass, varHandle.vmslot, testValue, newValue);
 /*[ELSE]
-				return _unsafe.weakCompareAndSwapObjectRelease(varHandle.definingClass, varHandle.vmslot, testValue, newValue);
+				return _unsafe.weakCompareAndSwapObjectRelease(definingClass, varHandle.vmslot, testValue, newValue);
 /*[ENDIF]*/				
 			}
 			
 			private static final boolean weakCompareAndSetPlain(Object testValue, Object newValue, StaticFieldVarHandle varHandle) {
+				Class<?> definingClass = varHandle.definingClass;
+				ensureClassIsInitialized(definingClass);
 /*[IF Sidecar19-SE-B174]*/
-				return _unsafe.weakCompareAndSetObjectPlain(varHandle.definingClass, varHandle.vmslot, testValue, newValue);
+				return _unsafe.weakCompareAndSetObjectPlain(definingClass, varHandle.vmslot, testValue, newValue);
 /*[ELSE]
-				return _unsafe.weakCompareAndSwapObject(varHandle.definingClass, varHandle.vmslot, testValue, newValue);
+				return _unsafe.weakCompareAndSwapObject(definingClass, varHandle.vmslot, testValue, newValue);
 /*[ENDIF]*/				
 			}
 			
 			private static final Object getAndSet(Object value, StaticFieldVarHandle varHandle) {
-				return _unsafe.getAndSetObject(varHandle.definingClass, varHandle.vmslot, value); 
+				Class<?> definingClass = varHandle.definingClass;
+				ensureClassIsInitialized(definingClass);
+				return _unsafe.getAndSetObject(definingClass, varHandle.vmslot, value); 
 			}
 			
 			private static final Object getAndSetAcquire(Object value, StaticFieldVarHandle varHandle) {
-				return _unsafe.getAndSetObjectAcquire(varHandle.definingClass, varHandle.vmslot, value); 
+				Class<?> definingClass = varHandle.definingClass;
+				ensureClassIsInitialized(definingClass);
+				return _unsafe.getAndSetObjectAcquire(definingClass, varHandle.vmslot, value); 
 			}
 			
 			private static final Object getAndSetRelease(Object value, StaticFieldVarHandle varHandle) {
-				return _unsafe.getAndSetObjectRelease(varHandle.definingClass, varHandle.vmslot, value); 
+				Class<?> definingClass = varHandle.definingClass;
+				ensureClassIsInitialized(definingClass);
+				return _unsafe.getAndSetObjectRelease(definingClass, varHandle.vmslot, value); 
 			}
 			
 			private static final Object getAndAdd(Object value, StaticFieldVarHandle varHandle) {
+				Class<?> definingClass = varHandle.definingClass;
+				ensureClassIsInitialized(definingClass);
```

Comment:
Is there a test case that shows that the `operationNotSupported()` cases should trigger initialization?  Applies here and throughout the file.

Reply:
A little eager with the "search-and-replace". I'll write a test case.

Reply:
I wrote a test case and ran it on HotSpot. It appears that HotSpot initializes the class on lookup, and not on invocation as the API specifies.

I've updated this implementation to not initialize the class if it's not used (i.e. if we throw an exception). This is already the behaviour when we throw `UnsupportedOperationException` for mutating access modes on final fields.